testing/tests/DevExpress.ui.widgets.dataGrid/dataGrid.tests.js

```diff
@@ -8871,6 +8870,34 @@ QUnit.test("Scrollbar should not be shown if column hiding is enabled and all co
     assert.equal(scrollable.$content().width(), scrollable._container().width(), "no scrollbar");
 });
 
+QUnit.test("Column hiding should work if the last not fixed column was hiden with redundant space when columnAutoWidth is true and columns has minWidth (T656342)", function(assert) {
+    // arrange
+    var columns,
+        adaptiveColumnWidth,
+        dataGrid = createDataGrid({
+            width: 200,
+            dataSource: [{ C0: 0, C1: 1, C2: 2 }],
+            columnHidingEnabled: true,
+            columnAutoWidth: true,
+            showColumnHeaders: false,
+            columns: [
+                { dataField: "C0", minWidth: 100, fixed: true },
+                { dataField: "C1", minWidth: 100 },
+                { dataField: "C2", minWidth: 100 }
+            ]
+        });
+
+    this.clock.tick();
+
+    columns = dataGrid.getController("columns").getColumns();
```

Comment:
The dataGrid.getVisibleColumns() will be better

Reply:
Sure. Corrected.